tests/unit/test_upstream_dataset.py

```diff
@@ -0,0 +1,110 @@
+import responses
+
+from tamr_unify_client import Client
+from tamr_unify_client.auth import UsernamePasswordAuth
+
+
+@responses.activate
+def test_upstream_dataset():
+
+    project_config = {
+        "name": "Project 1",
+        "description": "Mastering Project",
+        "type": "DEDUP",
+        "unifiedDatasetName": "Project 1 - Unified Dataset",
+        "externalId": "Project1",
+        "resourceId": "1",
+    }
+
+    project_datasets_json = {
+        "id": "unify://unified-data/v1/datasets/12",
+        "name": "Project_1_unified_dataset_dedup_features",
+        "description": "Features for all the rows and values in the source dataset. Used in Dedup Workflow.",
+        "version": "543",
+        "keyAttributeNames": ["entityId"],
+        "tags": [],
+        "created": {
+            "username": "admin",
+            "time": "2019-06-05T18:31:59.327Z",
+            "version": "212",
+        },
+        "lastModified": {
+            "username": "admin",
+            "time": "2019-07-18T14:19:28.133Z",
+            "version": "22225",
+        },
+        "relativeId": "datasets/12",
+        "upstreamDatasetIds": ["unify://unified-data/v1/datasets/8"],
+        "externalId": "Project_1_unified_dataset_dedup_features",
+    }
+
+    dataset_json = {
+        "id": "unify://unified-data/v1/datasets/12",
+        "name": "Project_1_unified_dataset_dedup_features",
+        "description": "Features for all the rows and values in the source dataset. Used in Dedup Workflow.",
+        "version": "543",
+        "keyAttributeNames": ["entityId"],
+        "tags": [],
+        "created": {
+            "username": "admin",
+            "time": "2019-06-05T18:31:59.327Z",
+            "version": "212",
+        },
+        "lastModified": {
+            "username": "admin",
+            "time": "2019-07-18T14:19:28.133Z",
+            "version": "22225",
+        },
+        "relativeId": "datasets/12",
+        "upstreamDatasetIds": ["unify://unified-data/v1/datasets/8"],
+        "externalId": "Project_1_unified_dataset_dedup_features",
+    }
+
+    upstream_json = ["unify://unified-data/v1/datasets/8"]
+
+    upstream_ds_json = {
+        "id": "unify://unified-data/v1/datasets/8",
+        "name": "Project_1_unified_dataset",
+        "description": "",
+        "version": "529",
+        "keyAttributeNames": ["tamr_id"],
+        "tags": [],
+        "created": {
+            "username": "admin",
+            "time": "2019-06-05T16:28:11.639Z",
+            "version": "83",
+        },
+        "lastModified": {
+            "username": "admin",
+            "time": "2019-07-22T20:31:23.968Z",
+            "version": "23146",
+        },
+        "relativeId": "datasets/8",
+        "upstreamDatasetIds": ["unify://unified-data/v1/datasets/6"],
+        "externalId": "Project_1_unified_dataset",
+    }
+
+    upstream_datasets = {
```

Comment:
This doesn't need to be a dictionary as you only use the relative id which you could get from upstream_ds_json